landlab/utils/tests/test_source_tracking_algorithm.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+"""
+Unit tests for source_tracking_algorithm.py
+@author: Sai Nudurupati & Erkan Istanbulluoglu
+"""
+
+from nose.tools import (assert_equal)
+import numpy as np
+
+from landlab import RasterModelGrid
+from landlab.components import FlowRouter
+from landlab.utils import (track_source,
+                           find_unique_upstream_hsd_ids_and_fractions)
+
```

Comment:
Recommend adding tests to verify that flow_directions can be specified multiple ways including that you throw the correct errors if flow_directions is misspecified.